### ISSUE TITLE
Sphinx' extract_metadata is deprecated

### ISSUE THREAD

**Issue by funkyfuture** [kalekundert/autoclasstoc#24]
when i build a documentation w/ Sphinx 5.2.3 this warning is emitted a lot:

```
…/site-packages/autoclasstoc/utils.py:170: RemovedInSphinx60Warning: extract_metadata() is deprecated.
```

since @haiiliin asked for a new release, i'd like to point out that i haven't had the time yet to look into this (maybe during this week), but would find it great when patch that addresses the reported behaviour was included in the release.

**Comment by kalekundert** [comment]
Thanks for letting me know about this, I'll try to look into it.

Also, I thought I had this repository configured to make a new release on each commit, but apparently that isn't working.  I'll have to look into that too...

Edit: I misread this at first: the patch is already in (see e66537a), there just needs to be a release.

**Comment by funkyfuture** [comment]
oh, to catch these things early, adding `-W error:::sphinx` to pytest's options should do the trick (it's untested for this concrete case).

> to make a new release on each commit

this doesn't seem advisable.

**Comment by haiiliin** [comment]
> Thanks for letting me know about this, I'll try to look into it.
> 
> Also, I thought I had this repository configured to make a new release on each commit, but apparently that isn't working. I'll have to look into that too...

Just for reference, it seems that the release will only be published when you push some commits to the `release` branch.
https://github.com/kalekundert/autoclasstoc/blob/d948ea79c431e2bdd689a976f2fa618fea1043d2/.github/workflows/test_and_release.yml#L32-L36
But I agree with @funkyfuture, it seems not a good idea.

**Comment by kalekundert** [comment]
> oh, to catch these things early, adding -W error:::sphinx to pytest's options should do the trick (it's untested for this concrete case).

That's a good idea, but this project has almost no automated tests, so it won't do much good at the moment.

> Just for reference, it seems that the release will only be published when you push some commits to the release branch.

Yeah, just figured that out.  I appreciate the concern about making releases too promiscuously, but at this point most of the contributions to this project are PRs, and I generally want to release those ASAP.  Plus, the commit messages control whether or not a release is made, so it's not really *every* commit.

---

Let me know if you still get the deprecation warning.  After looking at this more closely, it seems like the patch in question was included in the 1.5.0 release, which was made on 3/10/2022.  If you were using that release, there may still be something to fix.  But I'm going to close this issue for now.
